workbench-core/authorization/src/dynamicAuthorization/ddbDynamicAuthorizationPermissionsPlugin.ts

```diff
@@ -0,0 +1,165 @@
+    throw new Error('Method not implemented.');
+  }
+  public async getIdentityPermissionsBySubject(
+    getIdentityPermissionsBySubjectRequest: GetIdentityPermissionsBySubjectRequest
+  ): Promise<GetIdentityPermissionsBySubjectResponse> {
+    throw new Error('Method not implemented.');
+  }
+  public async createIdentityPermissions(
+    createIdentityPermissionsRequest: CreateIdentityPermissionsRequest
+  ): Promise<CreateIdentityPermissionsResponse> {
+    const { identityPermissions } = createIdentityPermissionsRequest;
+    //Check if there are 100 ops
+    if (identityPermissions.length > 100)
+      throw new ThroughputExceededError('Exceeds 100 identity permissions');
+
+    //Create an item with createIdenttiyPermissions
+    const putRequests = identityPermissions.map((identityPermission) => {
+      const item = this._transformIdentityPermissionsToItem(identityPermission);
+      const conditionExpression = 'attribute_not_exists(pk)';
+      return {
+        item,
+        conditionExpression
+      };
+    });
+
+    // Will fail entire transaction if a single one fails
+    try {
+      await this._dynamoDBService.commitTransaction({
+        addPutRequests: putRequests
+      });
+    } catch (err) {
+      if (err.name === 'TransactionCanceledException') {
+        throw new IdentityPermissionCreationError(`Failed due to an exception: ${err.message}`);
+      }
+      throw err;
+    }
+
+    return {
+      data: {
+        identityPermissions
+      }
+    };
+  }
+  public async deleteIdentityPermissions(
+    deleteIdentityPermissionsRequest: DeleteIdentityPermissionsRequest
+  ): Promise<DeleteIdentityPermissionsResponse> {
+    throw new Error('Method not implemented.');
+  }
+
+  /**
+   * @param identityPermission - {@link IdentityPermission} to be transformed
+   *
+   * Transforms data into this form
+   * ```
+   * {
+   *  "*pk*" : "<subjectType>|<subject_id>" // partition key
+   *  "*sk*": " <CRUD Action> | <effect> | <IdentityType>|<IdentityId>" // sort key
+   *  "identity": "<IdentityType>|<IdentityId>"
+   *  "effect": "ALLOW | DENY"
+   *  "action" : "<CRUD Action>"
+   *  "conditions": {}
+   *  "fields": []
+   *  "description": "Description of identity"
+   * }
+   * ```
+   * @returns
+   */
+  private _transformIdentityPermissionsToItem(
+    identityPermission: IdentityPermission
+  ): Record<string, JSONValue | Set<JSONValue>> {
+    const { subjectType, subjectId, action, effect, identityType, identityId } = identityPermission;
+    const pk = this._createIdentityPermissionsPartitionKey(subjectType, subjectId);
+    const sk = this._createIdentityPermissionsSortKey(action, effect, identityType, identityId);
+    const identity = `${identityType}|${identityId}`;
+    const conditions = identityPermission.conditions ?? {};
+    const fields = identityPermission.fields ?? [];
+    const description = identityPermission.description ?? '';
+
+    const item: Record<string, JSONValue | Set<JSONValue>> = {
```

Comment:
it looks like this object has a set type, is there a reason we went with a generic type over what is actually expected?